Jenkinsfile

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env groovy
+
+@Library('caf-continuous-integration') _
+
+// Configures the behavior of our stages.
+config = [
+    // Version dependency for the caf-continuous-integration library.
+    ciLibVersion: 1.0,
+    // GitHub path to repository.
+    repository: 'actor-framework/actor-framework',
+    // List of enabled checks for email notifications.
+    checks: [
+        'build',
+        'tests',
+    ],
+    // Default CMake flags by build type.
+    buildFlags: [
+        'CAF_ENABLE_ACTOR_PROFILER:BOOL=ON',
+        'CAF_ENABLE_EXAMPLES:BOOL=ON',
+        'CAF_ENABLE_RUNTIME_CHECKS:BOOL=ON',
+    ],
+    // Our build matrix. Keys are the operating system labels and values are build configurations.
+    buildMatrix: [
+        // Various Linux builds.
+        ['almalinux-8', [ // EOL: June 2029
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        ['centos-7', [ // EOL July 2024
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        ['debian-10', [ // EOL June 2024
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        ['debian-11', [ // EOL June 2026
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        ['fedora-37', [ // EOL December 2023
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        ['fedora-38', [ // EOL June 2024
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        ['ubuntu-20.04', [ // April 2025
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        ['ubuntu-22.04', [ // April 2027
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['release'],
+        ]],
+        // Extra debug build with exceptions disabled.
+        ['fedora-38', [
+            numCores: 4,
+            tags: ['docker'],
+            builds: ['debug'],
+            extraBuildFlags: [
+                'CAF_LOG_LEVEL:STRING=TRACE',
+                'CAF_ENABLE_ROBOT_TESTS:BOOL=ON',
+                'CAF_ENABLE_EXCEPTIONS:BOOL=OFF',
+                'CMAKE_CXX_FLAGS:STRING=-fno-exceptions',
+            ],
+        ]],
+        // Extra debug build for leak checking.
+        ['fedora-38', [
+            numCores: 4,
+            tags: ['docker', 'LeakSanitizer'],
+            builds: ['debug'],
+            extraBuildFlags: [
+                'CAF_LOG_LEVEL:STRING=TRACE',
+                'CAF_ENABLE_ROBOT_TESTS:BOOL=ON',
+                'CAF_SANITIZERS:STRING=address',
+            ],
+            extraBuildEnv: [
+                'ASAN_OPTIONS=detect_leaks=1',
+            ],
+        ]],
+        // Extra debug build with static libs, UBSan and hardening flags.
+        ['fedora-38', [
```

Comment:
Any way we could add some info on the build variant to the key found here. Like having the keys `fedora-38-static`, `fedora-38-noexcept` and `fedora-38-leak-check`. Here it is clear what is different from regular build, but when looking at Jenkins reports it would make more sense to see e.g. `fedora-38 passed` and `fedora-38-static` failed. 

If it means we have to keep multiple fedora-38 folders in the `.ci` directory then it's probably not worth it, but maybe something could be done in the `caf-continious-integration` library.

Reply:
I like that idea. The string you put there is used to generate the path for finding the `Dockerfile`. However, we could implement a separator that allows adding extra information. For example: `fedora-38:asan`. The lib could then simply ignore the `:asan` part when looking for the Dockerfile. What do you think?

I'd like to do that as a followup, though, to keep this PR focused on the "bringing back" part.

Reply:
The suffix separator would be easier to implement. Let's do that as some follow up then.